src/components/input/inputFileAvatar/inputFileAvatar.tsx

```diff
@@ -0,0 +1,184 @@
+import classNames from 'classnames';
+import { useCallback, useState } from 'react';
+import { ErrorCode, useDropzone, type FileRejection } from 'react-dropzone';
+import { Avatar } from '../../avatars';
+import { Icon, IconType } from '../../icon';
+import { Spinner } from '../../spinner';
+import { useInputProps } from '../hooks';
+import { InputContainer } from '../inputContainer';
+import { type IInputContainerAlert } from '../inputContainer/inputContainer.api';
+import { SelectState, type IInputFileAvatarProps } from './inputFileAvatar.api';
+
+export const InputFileAvatar: React.FC<IInputFileAvatarProps> = ({
+    onFileSelect,
+    maxFileSize = 0,
+    minDimension = 0,
+    maxDimension = 0,
+    onlySquare = true,
+    ...otherProps
+}) => {
+    const [selectState, setSelectState] = useState<SelectState>(SelectState.IDLE);
+    const [imagePreview, setImagePreview] = useState<string | null>(null);
```

Comment:
We can simply default it to undefined
```suggestion
    const [imagePreview, setImagePreview] = useState<string>();
```